cmd/access/node_builder/access_node_builder.go

```diff
@@ -760,7 +759,7 @@ func (builder *FlowAccessNodeBuilder) BuildExecutionSyncComponents() *FlowAccess
 					return nil, err
 				}
 
-				builder.ScriptExecutor = scripts
+				builder.ScriptExecutor.InitReporter(builder.ExecutionIndexer, scripts)
```

Comment:
Is it possible that we don't create the script executor until its dependencies are ready, this eliminates the case where ScriptExecutor's method is called before the indexer and scripts model are passed in.

Reply:
The challenge is that we want the API to run before the checkpoint loads. So there needs to be some way for the API to check if `scripts` is available. I used the approach of wrapping it in `engine/access/rpc/backend/script_executor.go`, but am open to other approaches if you have other ideas.

Reply:
+1 I've used the same approach for `GetRegisterValues`